AppinfoJsonFile.js

```diff
@@ -0,0 +1,309 @@
+        replace(/([^\\])\\'/g, "$1'").
+        replace(/^\\"/, '"').
+        replace(/([^\\])\\"/g, '$1"');
+
+    return unescaped;
+};
+
+/**
+ * Clean the string to make a resource name string. This means
+ * removing leading and trailing white space, compressing
+ * whitespaces, and unescaping characters. This changes
+ * the string from what it looks like in the source
+ * code but increases matching.
+ *
+ * @static
+ * @param {String} string the string to clean
+ * @returns {String} the cleaned string
+ */
+AppinfoJsonFile.cleanString = function(string) {
+    var unescaped = AppinfoJsonFile.unescapeString(string);
+
+    unescaped = unescaped.
+        replace(/\\[btnfr]/g, " ").
+        replace(/[ \n\t\r\f]+/g, " ").
+        trim();
+
+    return unescaped;
+};
+
+/**
+ * Make a new key for the given string. This must correspond
+ * exactly with the code in htglob jar file so that the
+ * resources match up. See the class IResourceBundle in
+ * this project under the java directory for the corresponding
+ * code.
+ *
+ * @private
+ * @param {String} source the source string to make a resource
+ * key for
+ * @returns {String} a unique key for this string
+ */
+AppinfoJsonFile.prototype.makeKey = function(source) {
+    return AppinfoJsonFile.unescapeString(source);
+};
+
+AppinfoJsonFile.prototype.loadSchema = function(source) {
+    var localizeProperties = {}, schemaFilePath;
+    if (this.project.schema) {
+        schemaFilePath = path.join(process.env.PWD, this.project.schema);
+    } else {
+        schemaFilePath = path.join(process.env.PWD, "node_modules", "ilib-loctool-webos-appinfo-json", "schema/appinfo.schema.json");
+    }
+    logger.debug("AppinfoJsonFileTyp load Schema File " + schemaFilePath + "?");
+    var loadSchemaFile, schemaData;
+
+    if (fs.existsSync(schemaFilePath)) {
+        loadSchemaFile = fs.readFileSync(schemaFilePath, "utf-8");
+        schemaData = JSON.parse(loadSchemaFile);
+    } else {
+        logger.warn("Could not open schema file: " + schemaFilePath);
+    }
+
+    for (var key in schemaData.properties) {
+        if (schemaData.properties[key].localizable == true) {
+            localizeProperties[key] = schemaData.properties[key];
+        }
+    }
+    return localizeProperties;
+};
+
+/**
+ * Parse the data string looking for the localizable strings and add them to the
+ * project's translation set.
+ * @param {String} data the string to parse
+ */
+AppinfoJsonFile.prototype.parse = function(data) {
+    logger.debug("Extracting strings from " + this.pathName);
+
+    this.parsedData = data;
+
+    if (typeof data !== "object") {
+        this.parsedData = JSON.parse(data);
+    }
+
+    if (!this.schema) {
+        this.schema = this.loadSchema();
+    }
+    this.resourceIndex = 0;
+    for (var property in this.schema) {
+        if ((this.parsedData[property]) &&
+            (this.schema[property].type === typeof this.parsedData[property])) {
```

Comment:
Should we add "&& this.schema[property].localizable" to the condition?

Reply:
When schema file loaded, it's already checked and only values that is true are extracted. https://github.com/iLib-js/ilib-loctool-webos-appinfo-json/blob/initialSetup/AppinfoJsonFile.js#L142. so only data type check is enough.